build.assets/grpcbox.mk

```diff
@@ -14,9 +14,14 @@ else
 GRPCBOX ?= $(GRPCBOX_BASE_NAME):$(BUILDBOX_VERSION)
 endif
 
+# UID and GID are used to run the grpcbox as the current user.
+UID := $$(id -u)
+GID := $$(id -g)
+
 # GRPCBOX_RUN has the necessary invocation to run a command inside the grpcbox.
 # Use this variable to run it from other Makefiles.
-GRPCBOX_RUN := docker run -it --rm -v "$$(pwd)/../:/workdir" -w /workdir $(GRPCBOX)
+# Set XDG_CACHE_HOME to let non-root user cache dependencies: https://github.com/bufbuild/buf/blob/f38ba9455c6650947b0b710327f9acb708da1196/private/pkg/app/app_unix.go#LL70C18-L70C32
+GRPCBOX_RUN := docker run -it --rm -u $(UID):$(GID) -e XDG_CACHE_HOME="/tmp/.cache" -v "$$(pwd)/../:/workdir" -w /workdir $(GRPCBOX)
```

Comment:
I'm tempted to only add the extra bindings on Linux, so there's less chance for side-effects elsewhere (although there is also an argument for always running like this).

Reply:
+1 for using it for macos as well, IMO.